### ISSUE TITLE
Improve performance of `Smt::sorted_pairs_to_leaves`

### ISSUE THREAD

**Issue by krushimir** [0xPolygonMiden/crypto#348]
`Smt::sorted_pairs_to_leaves` function uses a `current_leaf_buffer` vector to track key-value pairs corresponding to a single leaf. Consider refactoring this logic to use an enum as a buffer to  improve performance.

(from [#341 comment](https://github.com/0xPolygonMiden/crypto/pull/341#discussion_r1861427223))

**Comment by bobbinth** [comment]
@krushimir - is this addressed/resolved as a part of https://github.com/0xPolygonMiden/crypto/pull/365?

**Comment by krushimir** [comment]
No, addressed in a separate [PR](https://github.com/reilabs/miden-crypto/pull/2/) in a forked repo - can be merged to #365 though.

**Comment by krushimir** [comment]
@bobbinth this is now addressed/resolved in #365
